include/matx/executors/kernel.h

```diff
@@ -64,30 +65,43 @@ __global__ void matxOpT1Kernel(Op op, index_t size0) {
 
 template <class Op>
 __global__ void matxOpT2Kernel(Op op, index_t size0, index_t size1) {
-  index_t idx = static_cast<index_t>(blockIdx.x) * blockDim.x + threadIdx.x;
-  index_t idy = static_cast<index_t>(blockIdx.y) * blockDim.y + threadIdx.y;
-  if (idx < size1 && idy < size0) {
-    if constexpr (std::is_pointer_v<Op>) {
-      (*op)(idy, idx); 
+  
+  for(index_t idy = static_cast<index_t>(blockIdx.y) * blockDim.y + threadIdx.y;
```

Comment:
Do we want two different kernels just in case there's a perf difference, or keep the grid-stride-loop for everything?

Reply:
good idea.  fixed.